aws/elasticache.go

```diff
@@ -57,11 +120,20 @@ func nukeAllElasticacheClusters(session *session.Session, clusterIds []*string)
 
 	var deletedClusterIds []*string
 	for _, clusterId := range clusterIds {
-		params := elasticache.DeleteCacheClusterInput{
-			CacheClusterId: clusterId,
+		// First, we need to look up the cache cluster again to determine if it is a member of a replication group or not,
+		// because members of a replication group require that the replication group be deleted first
+		cacheCluster, describeErr := getSingleCacheCluster(svc, clusterId)
+		if describeErr != nil {
+			return describeErr
+		}
+
+		var err error
+		if cacheCluster.ReplicationGroupId == nil {
+			err = nukeNonReplicationGroupElasticacheCluster(svc, clusterId)
+		} else {
+			err = nukeReplicationGroupMemberElasticacheCluster(svc, cacheCluster)
```

Comment:
It looks like this will delete the replication group, but not the cluster. I think it needs to call `nukeNonReplicationGroupElasticacheCluster` after the replication group is deleted, right?

Reply:
That's what I thought as well, and I initially had it wired up that way, but was somewhat surprised that this lead to an error about the target cluster not being found. 

It looks like deleting the replication group actually deletes the member node as well.

Reply:
Ah gotcha! I think the comments need to be updated then, because the wording of the comment on line 124 ("because members of a replication group require that the replication group be deleted *first*", emphasis mine) made me think that the cache cluster needs to be deleted as well.

Reply:
Roger - done in [cc586f7](https://github.com/gruntwork-io/cloud-nuke/pull/335/commits/cc586f7e13b28067e1b8fe2a2c90f5ac112b520a)